api/src/main/java/io/druid/data/input/impl/NewSpatialDimensionSchema.java

```diff
@@ -76,13 +105,24 @@ public boolean equals(Object o)
 
     NewSpatialDimensionSchema that = (NewSpatialDimensionSchema) o;
 
-    return dims != null ? dims.equals(that.dims) : that.dims == null;
+    if (!super.equals(that)) {
```

Comment:
I've never seen `super.equals()` call in `equals()` implementations. This is fragile, e. g. if `DimensionSchema` changed and instead of `getClass() != that.getClass()` checks `DimensionSchema.class != that.getClass()`, which is also a popular pattern in `equals()` implementations (although it would probably be wrong for DimensionSchema, if it is designed for subclassing).

So I suggest either to add a test for `NewSpatialDimensionSchema.equals()`, so if `DimensionSchema` is changed unintentionally, the test will catch this error. Or, maybe, replace subclassing with composition.